### PR TITLE
Updates re differing params per row, dynamic tax lookup, error structure

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: docs/import_specifications.ADR.md [hunk 1]
```diff
@@ -135,43 +141,69 @@ checking, including for missing or unknown input parameters. Most error checking
...
+
+* `cannot_find_file` if an input file cannot be found
+* `cannot_parse_file` if an input file cannot be parsed
+* `unexpected_error` if some other error occurs
+
+The HTTP code returned will be:
+
+* 500 if all errors are `unexpected_error`
+* 404 if at least one error is `cannot_find_file`
+* 400 if at least one error is `cannot_parse_file`
```
Comment (MrCreosote) [inline docs/import_specifications.ADR.md:157]: This is going from "you're the least screwed" to "you're totally screwed". 500 implies you can try later when the server is fixed. 404 implies you can try later if you add the missing file. 400 means you're boned and you have to delete and start over.

**Review by briehl — COMMENTED** [review]
Just a clarification question for now. Seems fine, though.

File: docs/import_specifications.ADR.md [hunk 2]
```diff
@@ -135,43 +141,91 @@ checking, including for missing or unknown input parameters. Most error checking
...
+* `cannot_find_file` if an input file cannot be found
+* `cannot_parse_file` if an input file cannot be parsed
+* `multiple_specifications_for_data_type` if more than one tab or file per data type is submitted
+* `no_files_provided` if no files were provided
+* `unexpected_error` if some other error occurs
+
+The HTTP code returned will be:
+
+* 500 if all errors are `unexpected_error`
+* 404 if at least one error is `cannot_find_file`
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply docs/import_specifications.ADR.md]: Clarified

[the remark this answers is not in the document]
Comment (briehl) [reply docs/import_specifications.ADR.md]: 👍

**Review by briehl — APPROVED** [review]

**Pull request merged**